radixdlt/src/main/java/com/radixdlt/consensus/EventCoordinator.java

```diff
@@ -57,70 +64,100 @@ public EventCoordinator(
 		// only do something if we're actually the leader
-		if (!proposerElection.isValidProposer(self, round)) {
+		if (!proposerElection.isValidProposer(selfKey.getUID(), view)) {
 			return;
 		}
 
-		Vertex proposal = proposalGenerator.generateProposal(this.pacemaker.getCurrentRound());
+		Vertex proposal = proposalGenerator.generateProposal(this.pacemaker.getCurrentView());
 
 		// TODO: Handle empty proposals
 		if (proposal.getAtom() != null) {
 			this.networkSender.broadcastProposal(proposal);
+			log.info("Starting view " + view + " with proposal " + proposal);
 		}
 	}
 
 	public void processVote(Vote vote) {
-		// only do something if we're actually the leader for the next round
-		if (!proposerElection.isValidProposer(self, vote.getVertexMetadata().getRound().next())) {
-			log.warn(String.format("Ignoring confused vote %s for %s", vote.hashCode(), vote.getVertexMetadata().getRound()));
+		// only do something if we're actually the leader for the next view
+		if (!proposerElection.isValidProposer(selfKey.getUID(), vote.getVertexMetadata().getView().next())) {
+			log.warn(String.format("Ignoring confused vote %s for %s", vote.hashCode(), vote.getVertexMetadata().getView()));
 			return;
 		}
 
-		// accumulate votes into QCs
-		// TODO assumes a single node network for now
-		QuorumCertificate qc = new QuorumCertificate(vote, vote.getVertexMetadata());
-		this.safetyRules.process(qc);
+		// accumulate votes into QCs in store
+		Optional<QuorumCertificate> potentialQc = this.pendingVotes.insertVote(vote);
+		if (potentialQc.isPresent()) {
+			QuorumCertificate qc = potentialQc.get();
+			processQC(qc);
+		}
+	}
+
+	private void processQC(QuorumCertificate qc) {
+		// sync up to QC if necessary
 		this.vertexStore.syncToQC(qc);
-		this.pacemaker.processQC(qc.getRound())
-			.ifPresent(this::processNewRound);
+
+		// commit any newly committable vertices
+		this.safetyRules.process(qc)
+			.ifPresent(vertexId -> {
+				log.info("Committed vertex " + vertexId);
+
+				final Vertex vertex = vertexStore.commitVertex(vertexId);
+				final Atom committedAtom = vertex.getAtom();
+				mempool.removeCommittedAtom(committedAtom.getAID());
+			});
+
+		// start new view if pacemaker feels like it
+		this.pacemaker.processQC(qc.getView())
+			.ifPresent(this::processNewView);
 	}
 
-	public void processLocalTimeout(Round round) {
-		if (!this.pacemaker.processLocalTimeout(round)) {
+	public void processLocalTimeout(View view) {
+		if (!this.pacemaker.processLocalTimeout(view)) {
 			return;
 		}
 
-		this.networkSender.sendNewRound(new NewRound(round.next()));
+		try {
+			// TODO make signing more robust by including author in signed hash
+			ECDSASignature signature = this.selfKey.sign(Hash.hash256(Longs.toByteArray(view.next().number())));
+			this.networkSender.sendNewView(new NewView(selfKey.getPublicKey(), view.next(), signature));
+		} catch (CryptoException e) {
+			log.error("Failed to sign new view at " + view, e);
```

Comment:
This should probably be an IllegalStateException. If a node fails to sign then we don't want to continue as there are bigger problems.

Reply:
Yeah, that's a good idea